lib/rubocop/cop/style/empty_line_between_defs.rb

```diff
@@ -7,20 +7,26 @@ module Style
       # This cop checks whether method definitions are
       # separated by empty lines.
       class EmptyLineBetweenDefs < Cop
-        include OnMethodDef
         MSG = 'Use empty lines between method definitions.'.freeze
 
-        def on_method_def(node, _method_name, _args, _body)
-          return unless node.parent && node.parent.begin_type?
+        def on_begin(node)
+          prev = nil
```

Comment:
You could make this a bit more concise using `Enumerable#each_cons`, rather than manually tracking `prev`.